src/main/java/org/javarosa/core/model/trace/EvaluationTraceReporter.java

```diff
@@ -9,12 +9,28 @@
  *
  * Created by ctsims on 10/19/2016.
  */
-public interface EvaluationTraceReporter {
-    boolean wereTracesReported();
+public abstract class EvaluationTraceReporter {
 
-    void reportTrace(EvaluationTrace trace);
+    public abstract boolean wereTracesReported();
 
-    void reset();
+    public abstract void reportTrace(EvaluationTrace trace);
 
-    Vector<EvaluationTrace> getCollectedTraces();
+    public abstract void reset();
+
+    public abstract Vector<EvaluationTrace> getCollectedTraces();
+
+    public void printAndClearTraces(String description) {
```

Comment:
I don't think I'm following these reasons. Questions below:

1. "Moving this here will make him have to find a way to decouple the output at a class level rather than at the method call level." --> Doesn't having this be a method of an abstract class make it easier for Will to override its behavior, rather than harder? 

2. "One other huge advantage of the util method approach is that it lets us make internal decisions based on the type of trace reporter." --> The current approach makes it very straightforward to do that; subclasses would just have to override the base implementation of `printAndClearTraces()`